pypeit/coadd2d.py

```diff
@@ -893,11 +910,22 @@ def compute_offsets(self, offsets):
         Args:
             offsets (:obj:`list` or :obj:`str`):
                 Value that guides the determination of the offsets.
-                It could be a list of offsets, or a string, or None.
+                It could be a list of offsets, or a string.
 
         """
+        msgs.info('Get Offsets')
```

Comment:
I'm not following the inheritance structure here, but that could be a result of my ignorance. But it seems simpler to me to have routines like compute_offsets_init and compute_weights_init in the parent class. Those always execute the same code that is common to both multisite and echelle. Then compute_offsets could be called in the children without the need to use constructions like: 

super(EchelleCoAdd2D, self).compute_weights(weights)

But maybe your way is the more "object oriented" way to do it. To me it is confusing to see a method with the same name being overloaded in the child but still being called in the child.

Reply:
I would say this is standard practice.  E.g., we do this pretty frequently in the spectrograph classes.  We start by using the base-class method and then modify as necessary for the specific spectrograph.  I think this is fine and is, imo, cleaner than having "*_init" functions.

One comment on the syntax, though, in Python 3 you can just write, e.g., `super().compute_weights(weights)`.